document.go

```diff
@@ -0,0 +1,8 @@
+package smithy
```

Comment:
Is moving this to smithy package a temporary change? Or why should this be in smithy package rather than document package?

Reply:
I guess you answered this in PR description.

Reply:
Yeah after the design iterations, the basic `Document` type should stay in the smithy pkg. Concrete implementations probably will be in their own package, e.g. JSON and ION